From b57f53036d83c0658341ffd7ccfbff0544633a10 Mon Sep 17 00:00:00 2001
From: Shengqi Chen <harry-chen@outlook.com>
Date: Tue, 17 Dec 2024 01:40:41 +0800
Subject: [PATCH] simd_stat: fix undefined CONFIG_KERNEL_MODE_NEON error on
 armel

CONFIG_KERNEL_MODE_NEON depends on CONFIG_NEON. Neither is defined
on armel. Add a guard to avoid compilation errors.

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Shengqi Chen <harry-chen@outlook.com>
Closes #16871
---
 module/zcommon/simd_stat.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/module/zcommon/simd_stat.c b/module/zcommon/simd_stat.c
index 33c15140cdb9..d82a88ca9343 100644
--- a/module/zcommon/simd_stat.c
+++ b/module/zcommon/simd_stat.c
@@ -132,8 +132,10 @@ simd_stat_kstat_data(char *buf, size_t size, void *data)
 #if defined(__arm__) || defined(__aarch64__)
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
 		    "kernel_neon", HAVE_KERNEL_NEON);
+#if defined(CONFIG_KERNEL_MODE_NEON)
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
 		    "kernel_mode_neon", CONFIG_KERNEL_MODE_NEON);
+#endif /* CONFIG_KERNEL_MODE_NEON */
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
 		    "neon", zfs_neon_available());
 		off += SIMD_STAT_PRINT(simd_stat_kstat_payload,
